packages/dashboard/src/components/robots/robot-summary.tsx

```diff
@@ -0,0 +1,281 @@
+  TextField,
+  Theme,
+  Typography,
+} from '@mui/material';
+import { makeStyles, createStyles } from '@mui/styles';
+import React from 'react';
+import { RmfAppContext } from '../rmf-app';
+import { RobotTableData, base } from 'react-components';
+import { RobotState, Status2, TaskState } from 'api-client';
+import { EMPTY, combineLatest, mergeMap, of } from 'rxjs';
+import { TaskInspector } from '../tasks/task-inspector';
+import BatteryFullIcon from '@mui/icons-material/BatteryFull';
+import Battery60Icon from '@mui/icons-material/Battery60';
+import Battery20Icon from '@mui/icons-material/Battery20';
+import BatteryChargingFullIcon from '@mui/icons-material/BatteryChargingFull';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    textField: {
+      background: theme.palette.background.default,
+      '&:hover': {
+        backgroundColor: theme.palette.background.default,
+      },
+    },
+  }),
+);
+
+const setTaskDialogColor = (robotStatus: Status2 | undefined) => {
+  if (!robotStatus) {
+    return base.palette.background.default;
+  }
+
+  switch (robotStatus) {
+    case Status2.Error:
+      return base.palette.error.dark;
+
+    case Status2.Working:
+      return base.palette.success.dark;
+
+    default:
+      return base.palette.warning.main;
+  }
+};
+
+const LinearProgressWithLabel = (props: LinearProgressProps & { value: number }) => {
+  return (
+    <Box sx={{ display: 'flex', alignItems: 'center' }}>
+      <Box sx={{ width: '100%', mr: 1 }}>
+        <LinearProgress variant="determinate" {...props} />
+      </Box>
+      <Box sx={{ minWidth: 35 }}>
+        <Typography variant="body2" color="text.secondary">{`${Math.round(
+          props.value,
+        )}%`}</Typography>
+      </Box>
+    </Box>
+  );
+};
+
+interface RobotSummaryProps {
+  onClose: () => void;
+  robot: RobotTableData;
+}
+
+const showBatteryIcon = (robot: RobotState, robotBattery: number) => {
+  if (robot.status === Status2.Charging) {
+    return <BatteryChargingFullIcon />;
+  }
+
+  const batteryIcons: Record<string, JSX.Element> = {
+    '100': <BatteryFullIcon />,
+    '80': <Battery60Icon />,
+    '40': <Battery20Icon />,
+    '0': <Battery20Icon />,
+  };
+
+  const key = Object.keys(batteryIcons).find((level) => robotBattery <= parseInt(level));
+
+  return key ? batteryIcons[key] : null;
+};
+
+export const RobotSummary = React.memo(({ onClose, robot }: RobotSummaryProps) => {
+  const classes = useStyles();
+  const rmf = React.useContext(RmfAppContext);
+
+  const [isOpen, setIsOpen] = React.useState(true);
+  const [robotState, setRobotState] = React.useState<RobotState | null>(null);
+  const [taskState, setTaskState] = React.useState<TaskState | null>(null);
+  const [openTaskDetailsLogs, setOpenTaskDetailsLogs] = React.useState(false);
+  const [navigationStart, setNavigationStart] = React.useState('-');
+  const [navigationDestination, setNavigationDestination] = React.useState('-');
```

Comment:
Sorry for pingpong-ing on this, I just realized it would be difficult to maintain if the default values `-` are set in multiple locations, could we use a `string | null` for these states, and set it to null by default, as well as when we fail to retrieve the navigation starts and destinations.

Reply:
Nice!